pkgs/by-name/mg/mgitstatus/package.nix

```
@@ -0,0 +1,34 @@
+{ fetchFromGitHub, lib, stdenv, testers }:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mgitstatus";
+  version = "2.2";
+
+  src = fetchFromGitHub {
+    owner = "fboender";
+    repo = "multi-git-status";
+    rev = finalAttrs.version;
+    hash = "sha256-jzoX7Efq9+1UdXQdhLRqBlhU3cBrk5AZblg9AYetItg=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm 755 mgitstatus $out/bin/${finalAttrs.pname}
+    install -D mgitstatus.1 $out/man/man1/${finalAttrs.pname}.1
+
+    runHook postInstall
+  '';
+
+  passthru.tests.version = testers.testVersion { package = finalAttrs.pname; };
```

issue: This should be the derivation, not the `pname` (you need to pass `mgitstatus` to the arguments at the top of this file).

```suggestion
  passthru.tests.version = testers.testVersion { package = mgitstatus; };
```

---

Yeah, I was just testing this. Will fix in a sec.

---

@mrcjkb, I read through the docs and how a lot of the existing packages do tests, I cannot find if it's recommended to explicitly fill the `command` attr or not?

I tested the test and it works fine without explicitly stating the command but I'm unsure what's the convention regarding this.

---

It invokes `mainProgram --version` by default:

https://github.com/NixOS/nixpkgs/blob/32bc4a1b232a39fe3ad06a4cb7209ef92bd13ce9/pkgs/build-support/testers/default.nix#L58-L62